modules/@apostrophecms/page/index.js

```diff
@@ -1395,6 +1395,29 @@ database.`);
         if (self.isFound(req)) {
           return;
         }
+        if (req.user && (req.mode === 'published')) {
```

Comment:
This is how the second chance switch works. Server side, when it's about to 404, we look for whether the same URL would succeed in draft mode. if it would, we redirect with a query parameter as a hint that gets removed later via the history API so people don't share ugly URLs.